timelock-impl/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosTimestampBoundStore.java

```diff
@@ -289,6 +295,11 @@ private void checkAgreedBoundIsOurs(long limit, long newSeq, PaxosValue value) t
                 SafeArg.of("paxosValue", value));
     }
 
+    private synchronized void throwNotCurrentLeaderException(String message) {
```

Comment:
GaurdedBy annotation isn't smart enough to see that this is only called from synchronized methods.

Reply:
I'd normally be a bit anxious about the costs of re-acquiring the lock re-entrantly (and would check the cost of doing so - I always forget whether this can be considered cheap or not), but given this only happens in the unhappy path this should be ok.